src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -521,6 +598,43 @@ private String getCheckoutEffectiveCredentials() {
         }
     }
 
+    @NonNull
+    @Override
+    protected List<Action> retrieveActions(@CheckForNull SCMSourceEvent event,
+                                           @NonNull TaskListener listener)
+            throws IOException, InterruptedException {
+        // TODO when we have support for trusted events, use the details from event if event was from trusted source
+        List<Action> result = new ArrayList<>();
+        final BitbucketApi bitbucket = getBitbucketConnector().create(repoOwner, repository, getScanCredentials());
+        BitbucketRepository r = bitbucket.getRepository();
+        if (r != null) {
+            result.add(new BitbucketRepoMetadataAction(r));
+        }
+        String serverUrl = StringUtils.removeEnd(bitbucketServerUrl(), "/");
+        result.add(new BitbucketLink("icon-bitbucket-repo", serverUrl + "/" + repoOwner + "/" + repository));
+        return result;
+    }
+
+    @NonNull
+    @Override
+    protected List<Action> retrieveActions(@NonNull SCMHead head,
+                                           @CheckForNull SCMHeadEvent event,
+                                           @NonNull TaskListener listener)
+            throws IOException, InterruptedException {
+        // TODO when we have support for trusted events, use the details from event if event was from trusted source
+        List<Action> result = new ArrayList<>();
+        String serverUrl = StringUtils.removeEnd(bitbucketServerUrl(), "/");
+        String branchUrl;
+        if (head instanceof PullRequestSCMHead) {
+            PullRequestSCMHead pr = (PullRequestSCMHead) head;
+            branchUrl = pr.getRepoOwner() + "/" + pr.getRepository() + "/branch/" + pr.getBranchName();
+        } else {
+            branchUrl = repoOwner + "/" + repository + "/branch/" + head.getName();
```

Comment:
In the case of the pull request, linking to the pull request itself would make sense.

In the spots where Jenkins displays the COMMIT ID, such as the Changes page of a given job, I think linking to the specific Commit would be lovely.  

https://bitbucket.mycorp.com/projects/my-prj/repos/my-repo/commits/my-commit-id